### DISCUSSION TITLE
Issue: blocking mass requests

### DISCUSSION THREAD

**Discussion by ClubNZB** [spotweb/spotweb#870] [General]
Hi,

In the logfiles I see the follow:
X.X.X.1 - - [28/Dec/2024:12:52:47 +0100] "GET /?page=getnzb&action=display&messageid%5B%5D=cGRFM1Y1Q2JulcLkfmLexeYjqN2@spot.net&messageid%5B%5D=cGRFM1Y1Q2Juku5T2CSKHTnQfcf@spot.net&messageid%5B%5D=cGRFM1Y1Q2JuPtt26iCHWGmzpy7@spot.net&messageid%5B%5D=MXJZeEpDTFVuRtdbKNrkpPCyAVt@spot.net&messageid%5B%5D=WExxSmQzOXZ3yOxgVNRQ6QAzlf7@spot.net&messageid%5B%5D=WExxSmQzOXZ3UVFH1qywlQV42aP@spot.net&messageid%5B%5D=WExxSmQzOXZ3qEVvnn1zU7eRWd1@spot.net&messageid%5B%5D=WExxSmQzOXZ36CMuFZflBdGOZzS@spot.net&messageid%5B%5D=WExxSmQzOXZ3IXukWhQ2juMCFJv@spot.net&messageid%5B%5D=MXVxV3ZHelBQYEXCeysyLw2nq5e@spot.net&messageid%5B%5D=TFpWeEhUSlVUhXAX5V3ByZO6PxX@spot.net&messageid%5B%5D=TFpWeEhUSlVUPxjLGtKCnbJ8uBn@spot.net&messageid%5B%5D=cHNWT2hYenRik7PLxJCPPGGehtf@spot.net&messageid%5B%5D=dzRyUTRGRUtQjuRNSkSiNzCLgwR@spot.net&messageid%5B%5D=d29KVUdjV29zvl8dzuBL8Pbji5c@spot.net&messageid%5B%5D=ZXgxNXRpTDRaS9AvguzwptTjtYv@spot.net&messageid%5B%5D=ZXgxNXRpTDRaGHGSi2c3L0T2gWI@spot.net&messageid%5B%5D=T3diSlJaUzBP40QPGyjn8V13ZVR@spot.net&messageid%5B%5D=NlRoU2RKRXJlvvytBMqWxRayYRe@spot.net&messageid%5B%5D=Z244Z3JnNE1YhGftYF1OjnP8hGk@spot.net&messageid%5B%5D=Z244Z3JnNE1YzCO18qVipLXqPzg@spot.net&messageid%5B%5D=Z244Z3JnNE1YqUtJ42kBumxjuDz@spot.net&messageid%5B%5D=MUN5QnNPSGJKxvSWehJKVF2tqF5@spot.net&messageid%5B%5D=MUN5QnNPSGJKBwg9fuYxQNuP5wc@spot.net&messageid%5B%5D=MUN5QnNPSGJKAeTgkDWThMJbCgx@spot.net&messageid%5B%5D=b1kyeVZaU0hBFTzgujpBMqpJphH@spot.net&messageid%5B%5D=b1kyeVZaU0hBXcyOumOAi93Eaqg@spot.net&messageid%5B%5D=TkNjblRHdFh2byJxoBC3ZlxsHAH@spot.net&messageid%5B%5D=azhRQW51Nkp0yITQbuY88NK7FVm@spot.net&messageid%5B%5D=ZE9CSVJXZlJmTLTQwXmxNDtGdrh@spot.net&messageid%5B%5D=UmhUbFplcWFLZ9KbxNOQqrlLd3p@spot.net&messageid%5B%5D=eUd1MHowdm1hcsxyFtWyFpOfOlx@spot.net&messageid%5B%5D=eUd1MHowdm1hIFf85juoIf0LgVJ@spot.net HTTP/1.1" 502 564 "https://www.clubnzb.com/?page=index&search[tree]=cat0_z3_d74,cat0_z3_d24&search[value][]=NULL" "Mozilla/5.0 (Linux; Android 10; K) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Mobile Safari/537.36"

This request came back 20+ times a second and result the CPU and RAM of the VPS will go ski-high and slow down the website.
BitNinja can`t block this request, so is there a sollution to block this requests autmomaticly.
Somebody tips to put rechapta on the website from cloudflare or something to block this.

It is only "1 friend" who is doeing this, I block now manualy his KPN IP adress...

**Comment by Sweepr** [comment]
Seems like an NZB scraper-bot that uses VPN's, i have seen these too frequently and they only cause problems eat bandwidth, a real plague.
You could try to block them based on IP and setup a bot challenge, but once they change IP they are back, nothing much we can do about it.

**Comment by ClubNZB** [comment]
It is not a bot, it is a person who doesent like my website ore something.
It is always a KPN.com public IP address and the last 3 days not active. 
Now it is active, most of the time form KPN mobile.

there must be an option into papache or somewhere that only one messageid% can be requested

**Reply by Sweepr** [reply]
Programs like Hydra and Medusa can also be very resource hungry (give them a finger and they take your whole hand)

As a start i would look at how your server is configured, e.g. start with properly configuring memory limits for PHP if not already setup, normally a PHP process should not be allowed to use more memory that you allow it to.

From previous issue's i recall you have 16 gig RAM on your VPS? in order to have a process eat 16 gig's like a banana basically means something is misconfigured.

**Comment by mesa57** [comment]
The request always requests the same nzb's in succesive calls. Often 20 times in 1 second. This is clearly a ddos attack on the site.
I already pointed you to the apache module to prevent this.
As of selecting multiple nzb's : this is an option of the user interface. On the spotlist page you can select multiple nzb's which then will be combined by spotweb into 1 nzb and presented for download.
We are not going to disable that option,

**Reply by ClubNZB** [reply]
That apache module is installed, but no result Mesa.
And there is NO option to select multiple NZB`s, it is disabeld into the admin.

But thanks for your comment, but there is a reason why you don`t have access and I don`t need your help

**Reply by mesa57** [reply]
I am curious where that option is to disable select multiple nzb's
Please show a screenprint.

Besides that, also post you're mod_evasive.conf here.

**Reply by Sweepr** [reply]
> I am curious where that option is to disable select multiple nzb's Please show a screenprint.
> 
> Besides that, also post you're mod_evasive.conf here.

![image](https://github.com/user-attachments/assets/238beacf-81e8-4564-a7db-8af14adafd17)

It's: `$show_multinzb_checkbox = ($currentSession['user']['prefs']['show_multinzb']);`

**Reply by mesa57** [reply]
The checkbox present or not does not prevent spotweb from processing a request for multiple nzb's.
The ddos attacker crafted his request on beforehand and then just sends it to clubnzb. More then 20 times/sec.
